Updated Release Schedule Documentation (#4188)
dnnsoftware/Dnn.Platform: .github/RELEASE_SCHEDULE.md

# DNN Platform Release Schedule
To ensure adequate time for release planning by the community, partners, and vendors, a specific release process will be followed for all releases.  

## Release Candidates
Releases candidates (RC) are designed to give the community time to adjust their existing environments for any breaking changes and identify any unintended changes. Strong community participation during the RC process will result in more stable releases. 

> ### Major Releases (Example 10.0.0)

Author: Not sure I agree on this one, this still requires thinking

> ### Major Releases (Example 10.0.0)

Reviewer: @mitchelsellers this was just a way to keep it structured and so that we don't outdated it by giving random examples.  This is consistent with our Milestones too.

Major releases will have an initial RC cycle with a minimum duration of three-weeks from the date of RC release and final release. 

> ### Minor Releases (Example 9.8.x, or 9.9.x)

Reviewer 2: ```suggestion
### Minor Releases (Example 9.8.0, or 9.9.0)
```

Minor releases will have an initial RC cycle with a minimum duration of two weeks from the date of RC release and final release.

> ### Revision Releases (Example 9.8.1, or 9.8.2)

Reviewer 2: 👍🏻 Patch is the wording [the SemVer spec](https://semver.org/) uses.

Revision releases will have an initial RC cycle with a minimum duration of one week from the RC release and final release.

### Changes During RC Phase

> If necessary, changes will be incorporated during an RC; if the changes resolve significant issues or introduce risk, an additional RC may be created at the Approvers group's discretion.  If a revised RC is necessary, the Production Release schedule will be impacted; the exact impact will vary on a case-by-case basis depending on the nature of the issue(s) identified during the RC review. However, it will be communicated as part of the updated RC release notes.

Author: Really I guess "Final" should be used rather than "Production" here

> ## Production Releases

Reviewer: ```suggestion
## Official Releases
```

Author: Would "Final" work for you on this?  That matches the other locations

Reviewer: What other locations?  I am not really a fan of "Final", but I can change it if you all think it is the right label.

> If necessary, changes will be incorporated during an RC; if the changes resolve significant issues or introduce risk, an additional RC may be created at the Approvers group's discretion.  If a revised RC is necessary, the Final Release schedule will be impacted; the exact impact will vary on a case-by-case basis depending on the nature of the issue(s) identified during the RC review. However, it will be communicated as part of the updated RC release notes.

Reviewer: ```suggestion
If necessary, changes will be incorporated during an RC; if the changes resolve significant issues or introduce risk, an additional RC may be created at the discretion of the Approvers group.  If a revised RC is necessary, the Production Release schedule will be impacted; the exact impact will vary on a case-by-case basis depending on the nature of the issue(s) identified during the RC review. However, it will be communicated as part of the updated RC release notes.
```

> Production releases will only be completed after a successful RC phase, except in the case of a significant security release included as part of a revision release.

Reviewer: ```suggestion
Official releases will only be completed after a successful Release Candidate cycle, except in the case of a significant security release included as part of a Patch release.
```

> Final releases will only be completed after a successful RC phase, except in the case of a significant security release included as part of a revision release.

Reviewer: ```suggestion
Production Releases will only be completed after a successful Release Candidate cycle, except in the case of a significant security release included as part of a Patch release.
```

The anticipated release date will be communicated to the community at the time of the RC.  And each release will take the following considerations into mind for all releases.

> * Releases must allow for at least two additional business days after the release for regular Monday - Friday office situations (Releases only on Monday, Tuesday, or Wednesday)

Reviewer: ```suggestion
* Releases must allow for at least two business days following the release (based on standard business operations of Monday - Friday). Thus, releases should only be made on Mondays, Tuesdays, or Wednesdays.
```

* Releases will not be completed during weeks of major US holidays, specifically New Years, Memorial Day, Independence Day, Labor Day, Thanksgiving Day, or Christmas.
* Best efforts shall be made to avoid other significant holidays in other countries.

## Release Notifications

> You can utilize the "Watch" functionality within GitHub to receive notifications for new RC & Production releases using the "Releases Only" notification option.

Reviewer: ```suggestion
You can utilize the "Watch" functionality within GitHub to receive notifications for new Release Candidates and/or Official Releases using the "Releases Only" notification option.
```

Reviewer 2: It think it needs to be `and` rather than `and/or`, since you can't just subscribe to "official" releases.

Reviewer: Thanks - I updated.

> You can utilize the "Watch" functionality within GitHub to receive notifications for new Relese Candidate & Final releases using the "Releases Only" notification option.

Reviewer: ```suggestion
You can utilize the "Watch" functionality within GitHub to receive notifications for new Release Candidates and Production Releases using the "Releases Only" notification option.
```
